mypy/checker.py

```diff
@@ -3255,8 +3255,8 @@ def nothing() -> Iterator[None]:
 def is_typed_callable(c: Optional[Type]) -> bool:
     if not c or not isinstance(c, CallableType):
         return False
-    return not any(isinstance(t, AnyType) and t.implicit
-                   for t in c.arg_types + [c.ret_type])
+    return any(not isinstance(t, AnyType) or not t.implicit
```

Comment:
Hmm, yep, I think that's better!
I will make the change, make sure it passes the tests and then merge.